README.md

```diff
@@ -1,30 +1,205 @@
 # React Updating State
 
 ## Objectives
-
 1. Explain how to update state in React
-2. Explain the difference between changing state and changing props
-3. Describe what happens when state or props is updated
-4. Describe a React Event (SyntheticEvent)
+2. Describe what happens when state is updated
+3. Explain the difference between changing state and changing props
 
-## Overview
 
-This is where we dive in to `setState()` and that whole shebang. We can also
-introduce students to what happens when props change on an inner component.
+## Updating state
+![Unknown state](https://media.giphy.com/media/fFIaNdVhdvoOc/giphy.gif)
 
-I think we can continue to use some idea of a `Toggler` whose state is updated:
-that way we can just trigger an on/off switch and pass the result to the child
-components as a prop. (It's possible that I'm not seeing something gross about
-that implementation, since I'm just spitballing -- feel free to change it.)
+While a React component can have initial state, the real power is in updating its state — after all, if we didn't need
+to update the state, the component shouldn't _have_ any state. State is only reserved for data that _changes_ in our
+component and is visible in the UI.
 
-This is also the right time to start introducing students to forms and events.
-Definitely want to make use of a controlled component or two (which'll require
-state.)
+Instead of directly modifying the state using `this.state`, we use `this.setState()`. This is a function available to
+all React components, and allows us to let React know that the component state has changed. This way the components
+knows it should re-render, because its state has changed and its UI will most likely also change. Using a setter
+function like this is very performant: while other frameworks use dirty checking to see if a property has changed, React
```

Comment:
Perhaps put scare quotes around dirty checking to indicate that this is a term of art? Might also be nice to indicate which other frameworks use this method...